examples/README.md

```diff
@@ -114,3 +114,105 @@ To restore the snapshot to a new PVC, deploy
 kubectl create -f pvc-restore.yaml
 kubectl create -f pod-restore.yaml
 ```
+
+## How to enable multi node attach support for RBD
+
+*WARNING*  This feature is strictly for workloads that know how to deal
+with concurrent acces to the Volume (eg Active/Passive applications).
+Using RWX modes on non clustered file systems with applications trying
+to simultaneously access the Volume will likely result in data corruption!
+
+### Example process to test the multiNodeWritable feature
+
+Modify your current storage class, or create a new storage class specifically
+for multi node writers by adding the `multiNodeWritable: "enabled"` entry to
+your parameters.  Here's an example:
+
+```
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: csi-rbd
+provisioner: csi-rbdplugin
+parameters:
+    monitors: rook-ceph-mon-b.rook-ceph.svc.cluster.local:6789
+    pool: rbd
+    imageFormat: "2"
+    imageFeatures: layering
+    csiProvisionerSecretName: csi-rbd-secret
+    csiProvisionerSecretNamespace: default
+    csiNodePublishSecretName: csi-rbd-secret
+    csiNodePublishSecretNamespace: default
+    adminid: admin
+    userid: admin
+    fsType: xfs
+    multiNodeWritable: "enabled"
+reclaimPolicy: Delete
+```
+
+Now, you can request Claims from the configured storage class that include
+the `ReadWriteMany` access mode:
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-1
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: csi-rbd
+```
+
+Create a POD that uses this PVC:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-1
+spec:
+  containers:
+   - name: web-server
+     image: nginx
+     volumeMounts:
+       - name: mypvc
+         mountPath: /var/lib/www/html
+  volumes:
+   - name: mypvc
+     persistentVolumeClaim:
+       claimName: pvc-1
+       readOnly: false
+```
+
+Wait for the POD to enter Running state, write some data to
+`/var/lib/www/html`
+
+Now, we can create a second POD (ensure the POD is scheduled on a different
+node; multiwriter single node works without this feature) that also uses this
+PVC at the same time
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-2
+spec:
+  containers:
+   - name: web-server
+     image: nginx
+     volumeMounts:
+       - name: mypvc
+         mountPath: /var/lib/www/html
+  volumes:
+   - name: mypvc
+     persistentVolumeClaim:
+       claimName: pvc-1
+       readOnly: false
+```
+
+If you access the pod you can check that your data is avaialable at
```

Comment:
When the node with the second pod does the node PublishVolume it will run fsck against an active filesystem mounted on another node.  We'll get past this with XFS since (see ``fsck.xfs`` man page) it's a journaling filesystem that performs recovery at ``mount`` time instead so that ``fsck.xfs`` just exits with zero status.  But then node with the second pod will attempt the mount and run ``xfs_repair`` while the node with the first pod is writing to disk and journal.  We'll likely fail the mount and/or corrupt the filesystem at this point.  If by some miracle we got past this to actually writing to /var/lib/www/html we'd have two brains with their own journals trying to write to the same disk without coordinating with one another.

Reply:
That's not completely accurate; but regardless, keep in mind there is the multi-reader case that's very useful here as well.  Anyway, I get what you're saying and I'm not arguing really.  I am however saying that IMO the job of the storage plugin is to provide an interface to the device to expose it's capabilities and let users do what they want with them.  I'm working on a change to the multi-writer capability that I think will fit your view of the world better so hopefully you'll be happier with that one.  I'll be sure to ping you for reviews as soon as the PR is posted.

Reply:
The important part of this is whether I'm wrong that nodePublishVolume with volumeMode File and a node-local filesystem like ext4 or xfs has potential for corrupting the filesystem when it does fsck and mount from the second node.  If I'm mis-reading the code for that part then yeah, I'm being inaccurate and should have head adjusted with a 2x4.